test/links/connection/embed_id_test.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+class Chainer::Links::Connection::EmbedIDTest < Test::Unit::TestCase
+  data = {
+    test1: {x_data: [0, 1, 0], ignore_label: nil},
+    test2: {x_data: [[0, 1, 0], [1, 0, 1]], ignore_label: nil},
+    test3: {x_data: [0, 1, -1], ignore_label: -1},
+    test4: {x_data: [[0, 1, -1], [-1, 0, 1]], ignore_label: -1},
+  }
+
+  def _setup(data)
+    @link = Chainer::Links::Connection::EmbedID.new(3, 2, ignore_label: data[:ignore_label])
+    @link.cleargrads
+
+    @w = @link.w.data.copy()
+    @x = xm::Int32[*data[:x_data]]
+    y_shape = @x.shape + [2]
+    @gy = xm::SFloat.new(*y_shape).rand(-1, 1)
+  end
+
+  data(data)
+  def test_forward(data)
+    _setup(data)
```

Comment:
test-unit of the Red Chainer is currently used 3.29 or more.
From 3.29, it is now possible to refer to data even with setup.
ref. https://www.clear-code.com/blog/2018/12/26.html

please fix this.

Reply:
@hatappi I have fixed it!